core/src/test/java/org/jclouds/proxy/ProxyForURITest.java

```diff
@@ -41,12 +42,14 @@
 
    private class MyProxyConfig implements ProxyConfig {
       private boolean useSystem;
+      private boolean canUseProxyFromJvm;
       private Type type;
       private Optional<HostAndPort> proxy;
       private Optional<Credentials> credentials;
 
-      MyProxyConfig(boolean useSystem, Type type, Optional<HostAndPort> proxy, Optional<Credentials> credentials) {
+      MyProxyConfig(boolean useSystem, boolean canUseProxyFromJvm, Type type, Optional<HostAndPort> proxy, Optional<Credentials> credentials) {
```

Comment:
> it isn't whether it will use, it is whether it can use it.

Ah, then perhaps I'm misunderstanding the code. If this property is set, and no other properties override it, it **will** go through the default logic to pick a proxy? Or do you have to do something else to trigger that?

If setting this to/leaving it at true will cause a proxy to be configured (based on JVM properties) _even in the absence of other commands_, I would be tempted to interpret this is "**will** use the JVM proxy, if possible".

Don't want to make this a big thing, though - also OK with this if we feel it's more accurate.

@nacx Any particular thoughts from you on this?